doc/spring-boot-plugin-dev.md

```diff
@@ -0,0 +1,318 @@
+The context configuration contains the following XML content:
+
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+        http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+
+    <bean class="com.epimorphics.greeting.JsonLanguagePluginConfig" />
+</beans>
+```
+
+This loads the [JsonLanguagePluginConfig](https://github.com/simonoakesepimorphics/spring-plugin-example/blob/main/greeting-language-plugin/src/main/kotlin/com/epimorphics/greeting/JsonLanguagePluginConfig.kt) configuration class from the plugin JAR.
+
+To use multiple plugins at the same time, you must create a new XML configuration file containing the bean definitions of each.
+
+When we have written or chosen an external context configuration file for a particular deployment, we can proceed to running the application.
+
+## Running the Application with Plugins
+
+The standard Spring Boot application packaging builds an executable JAR containing the application and all of its dependencies, which is suitable for running the application alone.
+In order to insert one or more plugins into the application, we must run it with a customised main class and classpath;
+we will use the Spring Boot `PropertiesLauncher` class and the `-Dloader.path` argument for this purpose.
+See the [Spring documentation](https://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#appendix.executable-jar.launching) for more details.
+
+In summary, we must run the application with the following conditions:
+* Run the core application JAR with `java -cp`.
+* Add the plugin JAR to the classpath with `-Dloader.path`.
+* Use `org.springframework.boot.loader.PropertiesLauncher` as the main class.
+* Set the [external context configuraton](#externalized-configuration) location with the relevant configuration property (You can set this with either a program argument or an environment variable).
+
+For example, to run our example application with the language plugin:
+
+```
+CORE_APPLICATION_JAR=greeting-core/target/greeting-core-1.0-SNAPSHOT-exec.jar
+PLUGIN_JAR_WITH_DEPS=greeting-language-plugin/target/greeting-language-plugin-1.0-SNAPSHOT-jar-with-dependencies.jar
+EXTERNAL_PLUGIN_CONFIG=classpath:/context.xml
+
+java \
+  -cp $CORE_APPLICATION_JAR \
+  -Dloader.path=$PLUGIN_JAR_WITH_DEPS \
+  org.springframework.boot.loader.PropertiesLauncher \
+  --greeting.config.external.location=$EXTERNAL_PLUGIN_CONFIG
+```
+
+With variables being defined as follows:
+
+* `$CORE_APPLICATION_JAR` - The location of the executable application JAR.
+* `$PLUGIN_JAR_WITH_DEPS` - The location of the plugin library JAR with dependencies.
+* `$EXTERNAL_PLUGIN_CONFIG` - The location of the external context configuraton file or resource.
```

Comment:
I would drop the dollar sign before the variable name here as it is not part of the name